msteams-platform/task-modules-and-cards/cards/Universal-actions-for-adaptive-cards/Up-To-Date-Views.md

```diff
@@ -0,0 +1,241 @@
+---
+title: Up to date views
+description: Sample for up to date views using Universal Bot
+ms.topic: conceptual
+---
+
+## Up to date views
+Up to date views are now powered with a combination of Universal bots and message edits in Teams.
+
+Example: A user creates an asset approval request in Teams conversation. Shiladitya creates an approval request and assigns it to Sowrabh and Dipesh. 
+
+Part 1: Now role based views can be leveraged by using refresh property of the adaptive cards.
+Using role based views one can show a card with Approve/Reject buttons to a set of users, and show a card without Approve/Reject buttons to other users depending on their role.
+
+Part 2: To keep the card state updated an all times, Teams message edit mechanism can be leveraged.
+For example each time there is an approval, bot can trigger a message edit to all users. This bot message edit would trigger an `adaptiveCard/action` invoke request for all automatic refresh users, to which the bot can respond with the updated role based view card.
+
+**Sample approval base card**
+```JSON
+{
+  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+  "type": "AdaptiveCard",
+  "version": "1.4",
+  "refresh": {
+    "action": {
+      "type": "Action.Execute",
+      "title": "Refresh",
+      "verb": "roleBasedView"
+    },
+    "userIds": ["<Sowrabh's user MRI>", "<Dipesh's user MRI>"]
```

Comment:
Reg. `MRI`, please check existing documentation how it is referred to. Would be good to keep it consistent with other external facing documentation.